auraed/src/cells/cell_service/cell_service.rs

```diff
@@ -326,4 +366,93 @@ impl cell_service_server::CellService for CellService {
             self.stop_in_cell(&cell_name, request).await
         }
     }
+
+    async fn list(
+        &self,
+        _request: Request<CellServiceListRequest>,
+    ) -> std::result::Result<Response<CellServiceListResponse>, Status> {
+        Ok(Response::new(self.list().await?))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::cells::cell_service::validation::{
+        ValidatedCell, ValidatedCpuController, ValidatedCpusetController,
+    };
+    use iter_tools::Itertools;
+
+    // Ignored: requires sudo, which we don't have in CI
+    #[ignore]
+    #[tokio::test]
+    async fn test_list() {
+        let service = CellService::new();
+
+        let parent_cell_name = format!("ae-test-{}", uuid::Uuid::new_v4());
+        assert!(service
+            .allocate(allocate_request(&parent_cell_name))
+            .await
+            .is_ok());
+
+        let nested_cell_name =
+            format!("{}/ae-test-{}", &parent_cell_name, uuid::Uuid::new_v4());
+        assert!(service
+            .allocate(allocate_request(&nested_cell_name))
+            .await
+            .is_ok());
+
+        let cell_without_children_name =
+            format!("ae-test-{}", uuid::Uuid::new_v4());
+        assert!(service
+            .allocate(allocate_request(&cell_without_children_name))
+            .await
+            .is_ok());
+
+        let result = service.list().await;
+        assert!(result.is_ok());
+
+        let list = result.unwrap();
+        assert_eq!(list.cells.len(), 2);
+
+        let expected_root_cell_names =
+            vec![&parent_cell_name, &nested_cell_name].sort();
+        let actual_root_cell_names = list
+            .cells
+            .iter()
+            .map(|c| c.cell.as_ref().unwrap().name.clone())
+            .collect_vec()
+            .sort();
+        assert_eq!(actual_root_cell_names, expected_root_cell_names);
+
+        let parent_cell = list
+            .cells
+            .iter()
+            .find(|p| p.cell.as_ref().unwrap().name.eq(&parent_cell_name));
+        assert!(parent_cell.is_some());
+
+        let expected_nested_cell_names = vec![&nested_cell_name];
+        let actual_nested_cell_names = parent_cell
+            .unwrap()
+            .children
+            .iter()
+            .map(|c| c.cell.as_ref().unwrap().name.as_str())
+            .collect_vec();
+        assert_eq!(actual_nested_cell_names, expected_nested_cell_names);
+
+        _ = service.free_all().await;
```

Comment:
Is this call actually needed? It shouldn't be, so needing it indicates a bug somewhere.

Reply:
It is not. As we were doing the free-ing in the tests for `Cells` I copied that pattern here as I was trying to figure out why `make test-all` was flaky. Will remove this again as this didn't actually solve the problem.